app/controllers/cases/offender_sar_complaint_controller.rb

```diff
@@ -79,5 +79,15 @@ def process_date_responded_params
     def session_state
       "#{@correspondence_type_key}_state".to_sym
     end
+
+    private 
+
+    def clean_empty_approval_flags(permitted_params)
+      if permitted_params["approval_flag_ids"].present?
+        permitted_params["approval_flag_ids"].delete("")
```

Comment:
Not sure I follow this. It's saying `if value is present then delete the value with the key of "", or return nil if not found` is this right? Does the sub array have a key which is an empty string?

Reply:
When params returned from frontend contain the "approval_flag_ids",  it will include a empty "" in the array (not sure why),  this func is to remove this empty item from the array, otherwise a nil value will be stored although it won't cause any trouble, but just want to remove it